test/hotspot/gtest/runtime/test_signals.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#ifndef _WIN32
+#include "memory/allocation.hpp"
+#include "memory/resourceArea.hpp"
+#include "runtime/os.hpp"
+#include "utilities/defaultStream.hpp"
+#include "unittest.hpp"
+
+#include <signal.h>
+#include <stdio.h>
+#include <sys/ucontext.h>
+#include <string.h>
+
+extern "C" {
+  static void sig_handler(int sig, siginfo_t *info, ucontext_t *context) {
+    printf( " HANDLER (1) " );
+  }
+}
+
+class PosixSignalTest : public ::testing::Test {
+  public:
+
+  static const char* check_handlers() {
+    ResourceMark rm;
+    struct sigaction act, old_SIGFPE_act, old_SIGILL_act;
+    act.sa_handler = (void (*)(int))sig_handler;
+    sigemptyset(&act.sa_mask);
+    act.sa_flags = 0;
+    if (sigaction(SIGFPE, &act, &old_SIGFPE_act) == -1) {
```

Comment:
Instead of perror() which I haven't seen in the gtests (maybe it's ok), we usually use ASSERT statements like:
ASSERT_NE(sigaction(SIGFPE, &act, &old_SIGFPE_act), -1) << sigaction(SIGFPE) failed;

Reply:
I changed this to use ASSERT_*.